elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java

```diff
@@ -186,6 +186,11 @@ String scriptedFieldPrefix() {
     query.put("_source", false);
     query.put("size", 0);
     query.remove("script_fields");
+    // set _source = false and size = 0, `FetchPhase` would still be executed
+    // to fetch the metadata fields and visit the Lucene stored_fields,
+    // which would lead to performance declined dramatically.
+    // `stored_fields = _none` can prohibit such behavior entirely
+    query.put("stored_fields", "_none_");
```

Comment:
Sounds reasonable, do you think we can add a test case for this ? Some code to  illustrates that the `FetchPhase` you mentioned has indeed been triggered without your fix.

Reply:
hi, @danny0405  . Thanks for review.  I have worked on Doris On Elasticsearch (https://github.com/apache/incubator-doris/pull/2047).  I found this problem few months ago, If the query sent to Elasticsearch without `stored_fields : _none_` the fetch phase must be executed in order to get some meta data fields 

I submit a PR for es can implicitly promote this behavior: https://github.com/elastic/elasticsearch/pull/41155 

but they told me add `stored_fields : _none_`  to the query  can reach this goal.

Elasticsearch X-Pack SQL, they also add `stored_fields : _none_` to the query for Aggregation Transform

Reply:
All logic can see at `FetchPhase`:

https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

`FetchPhase#execute`

Reply:
Cool, thanks for the explanation ~